cryptofeed/exchange/binance_futures.py

```diff
@@ -79,14 +98,56 @@ async def _open_interest(self, msg: dict, timestamp: float):
                                 )
             self.open_interest[pair] = oi
 
+    async def _funding_with_index_price(self, msg: dict, timestamp: float):
+        """
+        {
+            "e": "markPriceUpdate",     // Event type
+            "E": 1562305380000,         // Event time
+            "s": "BTCUSDT",             // Symbol
+            "p": "11794.15000000",      // Mark price
+            "i": "11784.62659091",      // Index price
+            "P": "11784.25641265",      // Estimated Settle Price, only useful in the last hour before the settlement starts
+            "r": "0.00038167",          // Funding rate
+            "T": 1562306400000          // Next funding time
+        }
+        """
+        symbol = self.exchange_symbol_to_std_symbol(msg['s'])
+        exchange_index_symbol = self._translate_index_symbol(msg['s'], False)
+        index_symbol = self.exchange_symbol_to_std_symbol(exchange_index_symbol)
+        ts = timestamp_normalize(self.id, msg['E'])
+
+        if self._subscribed_to_feed_and_symbol(FUNDING, msg['s']):
+            # Subscribed to the non-index symbol and FUNDING. Send an update.
+            await self.callback(FUNDING,
+                                feed=self.id,
+                                symbol=symbol,
+                                timestamp=ts,
+                                receipt_timestamp=timestamp,
+                                mark_price=msg['p'],
+                                rate=msg['r'],
+                                next_funding_time=timestamp_normalize(self.id, msg['T']),
+                                )
+
+        if self._subscribed_to_feed_and_symbol(UNDERLYING_INDEX, exchange_index_symbol):
+            # Subscribed to the index symbol and UNDERLYING_INDEX. Send an update.
+            await self.callback(UNDERLYING_INDEX, feed=self.id,
+                            symbol=index_symbol,
+                            timestamp=ts,
+                            receipt_timestamp=timestamp,
+                            price=msg['i'])
+
     def connect(self) -> List[Tuple[AsyncConnection, Callable[[None], None], Callable[[str, float], None]]]:
         ret = []
         if self.address:
             ret = super().connect()
 
         for chan in set(self.subscription):
             if chan == 'open_interest':
-                addrs = [f"{self.rest_endpoint}/openInterest?symbol={pair}" for pair in self.subscription[chan]]
+                addrs = []
+                for pair in self.subscription[chan]:
+                    if pair.startswith(INDEX_PREFIX):
+                        continue
+                    addrs.append(f"{self.rest_endpoint}/openInterest?symbol={pair}")
```

Comment:
this is absolutely fine but it is usually more pythonic to use list comprehension

```
addrs = [
    f"{self.rest_endpoint}/openInterest?symbol={pair}" 
    for pair in self.subscription[chan] if not pair.startswith(INDEX_PREFIX)
]
```